Lib/src/main/java/uk/ac/cam/cl/wildpetscience/triton/lib/pipeline/tracker/ClusteringModule.java

```diff
@@ -0,0 +1,162 @@
+package uk.ac.cam.cl.wildpetscience.triton.lib.pipeline.tracker;
+
+import org.opencv.core.Mat;
+import org.opencv.core.Size;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+import static org.opencv.imgproc.Imgproc.*;
+
+/**
+ * Algorithm that, given an image depicting areas of motion
+ * outputs the center of motion of the largest cluster
+ *
+ * Created by Vlad on 19/02/2015.
+ */
+
+//
+public class ClusteringModule {
+    private double prob;
+    private boolean[][] used, movement;
+
+    private int area, maxArea, totalArea;
+    private int rows, cols;
+    private final int vicinity = 1;
+
+    private int xCurr, yCurr, xAns, yAns;
+
+    private Mat image;
+
+    public ClusteringModule(Mat img) {
+        totalArea = 0;
+        prob = 0.0;
+        rows = img.rows();
+        cols = img.cols();
+
+        xAns = 0;
+        yAns = 0;
+        image = img;
+
+        if(rows == 0 || cols == 0)
+            return;
+
+        used = new boolean[rows][cols];
```

Comment:
rows and cols aren't going to change often - is it possible to cache these? Otherwise we are allocating a massive amount of heap on every frame. Perhaps make this class into a singleton, then have a process() method that does all of this, but then re-allocates `used` and `movement` if the rows and cols have changed.